### PR TITLE
fix(rhino): Fixed regression converting elements from connector bindings

### PR DESCRIPTION
**Pull request by JR-Morgan**
Fixed the regression introduced by #1916

I've commented the code with a short explanation of the problem

targeting #1922 for @clairekuang to review

### CODE REVIEW

File: ConnectorRhino/ConnectorRhino/ConnectorRhinoShared/UI/ConnectorBindingsRhino.cs [hunk 1]
```diff
@@ -386,32 +389,41 @@ public override async Task<StreamState> ReceiveStream(StreamState state, Progres
...
-        // flatten the commit object to retrieve children objs
-        int count = 0;
-        Preview = FlattenCommitObject(commitObject, converter, progress, commitLayerName, ref count);
-
-        // convert
-        foreach (var previewObj in Preview)
+        // create preview objects if they don't already exist
+        if (Preview.Count == 0)
         {
-            var storedObj = StoredObjects[previewObj.OriginalId];
```
Comment (JR-Morgan) [inline ConnectorRhino/ConnectorRhino/ConnectorRhinoShared/UI/ConnectorBindingsRhino.cs:399]: This line was sometimes throwing an exception, as `previewObj.OrianlId` is not in `StoredObjects`

Previously (rc3), this lookup wouldn't happen until Convert was called. And wouldn't happen for paths where the object is not convertible.

File: ConnectorRhino/ConnectorRhino/ConnectorRhinoShared/UI/ConnectorBindingsRhino.cs [hunk 2]
```diff
@@ -386,32 +389,41 @@ public override async Task<StreamState> ReceiveStream(StreamState state, Progres
...
+          Preview = FlattenCommitObject(commitObject, converter, progress, commitLayerName, ref count);
+
+          // convert
+          foreach (var previewObj in Preview)
+          {
+            if (previewObj.Convertible)
             {
-              previewObj.Update(status: ApplicationObject.State.Failed, logItem: $"Couldn't retrieve stored object from bindings");
-              continue;
+              var storedObj = StoredObjects[previewObj.OriginalId];
```
Comment (JR-Morgan) [inline ConnectorRhino/ConnectorRhino/ConnectorRhinoShared/UI/ConnectorBindingsRhino.cs:404]: Moving this line inside the `if(previewObj.Convertable)` block fixes this!

File: ConnectorRhino/ConnectorRhino/ConnectorRhinoShared/UI/ConnectorBindingsRhino.cs [hunk 3]
```diff
@@ -238,6 +238,9 @@ public override void ResetDocument()
 
     #region receiving 
     public override bool CanPreviewReceive => true;
+
+    private static bool IsPreviewIgnore(Base @object) => @object.speckle_type.Contains("Block") || @object.speckle_type.Contains("View");
```
Comment (JR-Morgan) [inline ConnectorRhino/ConnectorRhino/ConnectorRhinoShared/UI/ConnectorBindingsRhino.cs:242]: additionally, I noticed that this check was being performed in 3 places. so extracted a method.

**Comment by clairekuang** [comment]
ooh ty for the catch!

**Pull request merged**